src/borg/crypto/key.py

```diff
@@ -406,18 +410,33 @@ def init_ciphers(self, manifest_nonce=0):
 
 class Passphrase(str):
     @classmethod
-    def _env_passphrase(cls, env_var, default=None):
+    def _get_env(cls, env_var, default=None):
         passphrase = os.environ.get(env_var, default)
         if passphrase is not None:
             return cls(passphrase)
 
     @classmethod
     def env_passphrase(cls, default=None):
-        return cls._env_passphrase('BORG_PASSPHRASE', default)
+        passphrase = cls._get_env('BORG_PASSPHRASE', default)
+        if passphrase is not None:
+            return passphrase
+        passphrase = cls.env_passcommand()
+        if passphrase is not None:
+            return passphrase
+
+    @classmethod
+    def env_passcommand(cls, default=None):
+        passcommand = cls._get_env('BORG_PASSCOMMAND', default)
```

Comment:
that's a bit weird. the returned object is of class `Passphrase` (if not None), but does contain the command, not a passphrase. so i guess you'ld rather call `os.environ.get('BORG_PASSCOMMAND', default)` here.